test/e2e/autoscaling/custom_metrics_stackdriver_autoscaling.go

```diff
@@ -54,6 +55,15 @@ var _ = SIGDescribe("[HPA] Horizontal pod autoscaling (scale resource: Custom Me
 	f := framework.NewDefaultFramework("horizontal-pod-autoscaling")
 	f.NamespacePodSecurityEnforceLevel = admissionapi.LevelPrivileged
 
+	// Set up FeatureGate for enabling HPAScaleToZero, which is hidden by alpha feature gate.
+	fg := featuregate.NewFeatureGate()
+	fg.Add(map[featuregate.Feature]featuregate.FeatureSpec{"HPAScaleToZero": {
+		Default:       false,
+		LockToDefault: false,
+		PreRelease:    featuregate.Alpha,
+	}})
```

Comment:
is it normal to have to re-define feature gates in the e2e binary? what keeps this from drifting from the default enablement in kube_features.go?

Reply:
I've refactored this, introducing local `withFeatureGate` method, which is re-implementing logic used by `featuregatetesting.SetFeatureGateDuringTest` - as these ginkgo e2e tests don't have testing context available (as e.g. integration tests)